source/OutfitterPanel.cpp

```diff
@@ -330,427 +362,573 @@ double OutfitterPanel::DrawDetails(const Point &center)
+			if(shipsToOutfit.empty())
+				break;
+			for(Ship *ship : shipsToOutfit)
+			{
+				// Uninstall the outfit.
+				ship->AddOutfit(selectedOutfit, -1);
+				int required = selectedOutfit->Get("required crew");
+				if(required)
+					ship->AddCrew(-required);
+				// Adjust hired crew counts.
+				ship->Recharge();
+
+				// If the context is sale:
+				if(toLocation == OutfitLocation::Shop)
 				{
-					ship->AddOutfit(ammo, -mustSell);
-					if(toStorage)
-						mustSell -= storage.Add(ammo, mustSell);
-					if(mustSell)
+					// Do the sale.
+					int64_t price = player.FleetDepreciation().Value(selectedOutfit, day);
+					player.Accounts().AddCredits(price);
+					player.AddStock(selectedOutfit, 1);
+				}
+				// If the context is uninstalling, move the outfit into Storage
+				else if(toLocation == OutfitLocation::Storage)
+					// Move to storage.
+					player.Storage().Add(selectedOutfit, 1);
+				// Note: It would be easy to add conditional statements above to also support uninstall into cargo,
+				//		 this is not supported in the outfitter at this time.
+
+				// Move ammo to storage.
+				// Since some outfits have ammo, remove any ammo that must also be moved as there
+				// aren't enough supporting slots for said ammo once this outfit is removed.
+				const Outfit *ammo = selectedOutfit->AmmoStoredOrUsed();
+				if(ammo && ship->OutfitCount(ammo))
+				{
+					// Determine how many of this ammo we must uninstall to also uninstall the launcher.
+					int mustUninstall = 0;
+					for(const pair<const char *, double> &it : ship->Attributes().Attributes())
+						if(it.second < 0.)
+							mustUninstall = max<int>(mustUninstall, it.second / ammo->Get(it.first));
+
+					if(mustUninstall)
 					{
-						int64_t price = player.FleetDepreciation().Value(ammo, day, mustSell);
-						player.Accounts().AddCredits(price);
-						player.AddStock(ammo, mustSell);
+						ship->AddOutfit(ammo, -mustUninstall);
+
+						// If the context is sale:
+						if(toLocation == OutfitLocation::Shop)
+						{
+							// Do the sale of the outfit's ammo.
+							int64_t price = player.FleetDepreciation().Value(ammo, day, mustUninstall);
+							player.Accounts().AddCredits(price);
+							player.AddStock(ammo, mustUninstall);
+						}
+						// If the context is uninstalling, move the outfit's ammo into Storage
+						else if(toLocation == OutfitLocation::Storage)
+							// Move to storage.
+							player.Storage().Add(ammo, mustUninstall);
+						// Note: It would be easy to add conditional statements above to also support uninstall into
+						//		 cargo, this is not supported in the outfitter at this time.
 					}
 				}
 			}
 		}
-		return;
+		// Note: Uninstalling into cargo could be implemented below, but not supported in current outfitter logic.
 	}
-
-	if(!toStorage && storage.Get(selectedOutfit))
+	else if(fromLocation == OutfitLocation::Storage || fromLocation == OutfitLocation::Cargo)
```

Comment:
Well, what other options do we have at this point?
```suggestion
	else
```

Reply:
Done. (left as-is)